ethereum-network-partition-test/main.star

```diff
@@ -0,0 +1,123 @@
+eth2 = import_module("github.com/kurtosis-tech/example/ethereum-network-partition-test/eth2_starlark_package.star")
+ethereum_helpers = import_module("github.com/kurtosis-tech/example/ethereum-network-partition-test/ethereum_helpers.star")
+
+NUM_PARTICIPANTS = 4
+
+MAIN_NETWORK = "main_network"
+ISOLATED_NETWORK = "isolated_network"
+
+CHECKPOINT_1_NODES_SYNCED = "0x2" # block number 2
+CHECKPOINT_2_NODES_OUT_OF_SYNC = "0xc" # block number 12
+CHECKPOINT_3_NODES_SYNCED = "0x16" # block number 22
+
+
+def run(plan):
+    plan.print("Spinning up an Ethereum network with '{0}' nodes".format(NUM_PARTICIPANTS))
+    eth2.run_eth2_package(plan, NUM_PARTICIPANTS)
+
+    network_topology = assign_nodes_subnetwork(plan)
+    plan.print("Starting test with the following network topology: \n{0}".format(network_topology))
+
+    assert_all_nodes_synced_at_block(plan, CHECKPOINT_1_NODES_SYNCED)
+    plan.print("Block number '{0}' for all services: \n{1}".format(CHECKPOINT_1_NODES_SYNCED, get_block_all_nodes(plan, CHECKPOINT_1_NODES_SYNCED)))
+
+    plan.set_connection((MAIN_NETWORK, ISOLATED_NETWORK), kurtosis.connection.BLOCKED)
+    plan.print("Subnetwork '{0}' was disconnected from '{1}'".format(ISOLATED_NETWORK, MAIN_NETWORK))
+
+    assert_nodes_out_of_sync_at_block(plan, CHECKPOINT_2_NODES_OUT_OF_SYNC)
+    plan.print("Block number '{0}' for all services: \n{1}".format(CHECKPOINT_2_NODES_OUT_OF_SYNC, get_block_all_nodes(plan, CHECKPOINT_2_NODES_OUT_OF_SYNC)))
+
+    plan.remove_connection((MAIN_NETWORK, ISOLATED_NETWORK))
+    assert_all_nodes_synced_at_block(plan, CHECKPOINT_3_NODES_SYNCED)
+    plan.print("Block number '{0}' for all services: \n{1}".format(CHECKPOINT_3_NODES_SYNCED, get_block_all_nodes(plan, CHECKPOINT_3_NODES_SYNCED)))
+
+def assign_nodes_subnetwork(plan):
+    """
+    Assigns half of the nodes to subnetwork MAIN_NETWORK and the other half to ISOLATED_NETWORK.
+
+    The connection between the 2 subnetworks are not updated in this function
+    """
+    all_nodes = []
+    for i in range(0, NUM_PARTICIPANTS):
+        # each ETH node is composed of a tuple (EL, CL_VALIDATOR, CL_BEACON)
+        # We need to update the three of them
+        all_nodes.append((
```

Comment:
i haven't thought through this enouguh but but I don't think all eth clients end up having 2 cl nodes and one eth node, I guess we support only geth/lighthouse?

Reply:
yes, I just looked at code above and confirmed we are limited to geth/lighthouse for this test and we don't take user supplied args, so its alright 👍 

maybe put in a notice here that it might break for others ?

Reply:
Yes, this example supports only `geth/lighthouse`. FWIW it was the same for the Go equivalent. Maybe we could support all of them, but too much effort right now :D